_posts/pl/2024-04-19-czy-wiesz-jak-uzywac-operatorow-w-javascripcie-i-typescripcie.md

```diff
@@ -0,0 +1,90 @@
+---
+layout:    post
+title:     Czy wiesz, jak używać operatorów "?" i "!" w JavaScripcie i TypeScripcie?
+date:      2024-04-19T08:00:00+01:00
+published: true
+didyouknow: true
+lang: pl
+author: pgrobelny
+image: /assets/img/posts/2024-04-19-czy-wiesz-jak-uzywac-operatorow-w-javascripcie-i-typescripcie/question-marks.webp
+tags:
+- javascript
+- typescript
+---
+
+Wykrzykniki i znaki zapytania mogą czasami wprawić w zakłopotanie, ponieważ dalece im w kodzie od swojego interpunkcyjnego znaczenia.
+
+Na początek spójrzmy na wykrzyknik. W TypeScripcie może być on używany jako tzw. non-null assertion operator. Oznacza to, że znak `!` możemy potraktować jako informację dla kompilatora, że dana zmienna nie będzie miała wartości `null` ani `undefined` w momencie użycia. 
```

Comment:
W TypeScript może być używany jako tzw. non-null assertion operator. Oznacza to, że znak `!` możemy potraktować jako informację dla kompilatora, że dana zmienna nie będzie miała wartości `null` ani `undefined` w momencie użycia.